### ISSUE TITLE
Parcel has removed the shouldReload config api

### ISSUE THREAD

**Issue by JoshAshby** [orlov-vo/parcel-transformer-svelte#16]
Hey there, it looks like Parcel has recently removed the `config.shouldReload()` method for [the config object recently](https://github.com/parcel-bundler/parcel/commit/806cc65e0e430b1120cdcf1676e2f58738326d30#diff-9a32c845edc94e150db130efb9e733657ca2092b7d6fc1b53570a78c22190fd4L20) which seems to break this extension on recent nightly versions of Parcel (looks like any version past `2.0.0-nightly.610`) as a result of [`loadConfig.js`](https://github.com/orlov-vo/parcel-transformer-svelte/blob/ba7a6449fcdddc9a021c13cf8737d387ec02bdee/loadConfig.js#L46) using the `shouldReload()` method.

`svelte.config.js`:

```
const sveltePreprocess = require("svelte-preprocess")

module.exports = {
  preprocess: sveltePreprocess(),
}
```

`.parcelrc`:

```json
{
  "extends": ["@parcel/config-default"],
  "transformers": {
    "*.svelte": ["parcel-transformer-svelte"]
  }
}
```

Error with a recent parcel version (2.0.0-nightly.619 in this case):

```
$ parcel src/manifest.json
🚨 Build failed.
parcel-transformer-svelte: config.shouldReload is not a function
TypeError: config.shouldReload is not a function
    at load ([...]/node_modules/parcel-transformer-svelte/loadConfig.js:46:14)
    at async Transformation.loadTransformerConfig ([...]/node_modules/@parcel/core/lib/Transformation.js:739:7)
    at async Transformation.loadPipeline ([...]/node_modules/@parcel/core/lib/Transformation.js:677:20)
    at async Transformation.run ([...]/node_modules/@parcel/core/lib/Transformation.js:315:20)
    at async Child.handleRequest ([...]/node_modules/@parcel/workers/lib/child.js:255:9)
```

**Comment by zhyc9de** [comment]
same issue

**Comment by leeknowlton** [comment]
+1

Would replacing `config.shouldReload()` with `config.getConfig` work?

**Comment by schuetzm** [comment]
The commit at https://github.com/schuetzm/parcel-transformer-svelte/commit/fixes seems to work, and an additional error with `shouldInvalidateOnStartup()`. But I don't really understand the code, so it could be wrong.

As for `config.shouldReload()`, the [commit that removed it](https://github.com/parcel-bundler/parcel/commit/806cc65e0e430b1120cdcf1676e2f58738326d30) replaced it with a call to `config.addDevDependency()`.

**Comment by orlov-vo** [comment]
Yeah, it seems `shouldReload` is not required anymore. I've released the fix in v1.2.3 to npm
